### DISCUSSION TITLE
Improve documentation with real-life/practical examples

### DISCUSSION THREAD

**Discussion by stephannv** [MikeSchulze/gdUnit4#232] [Ideas]
Hi, I'm trying to getting into Godot testing and found gdUnit4. The docs are great, explains well all gdUnit core features. But I'm having a hard time to think how to use gdUnit on a real-life project (I have a lot of experience testing web applications but 0 xp testing games). For example: I know how to test a simple gdscript object, methods, returns, properties, mocks, etc, but how to test my player class jump method is correct? How do I test that a object followed a specific path and played some music and enabled some button? 
I don't know if these cases are on gdUnit scope, since these cases looks like integration/e2e tests, and gdUnit seems to be, even by Unit on name, just for unit testing.

**Comment by MikeSchulze** [comment]
Hello and welcome to GdUnit4.
For integration testing, you should use the built-in [GdUnitSceneRunner](https://mikeschulze.github.io/gdUnit4/advanced_testing/sceneRunner/#scene-runner) to test your game scenes.

There are some examples for GdUnit3 https://github.com/MikeSchulze/gdUnit3-examples, I haven't had time to migrate them to GdUnit4 yet.
But it should give you a good guide on how to test game scenes.

**Reply by stephannv** [reply]
Thanks @MikeSchulze . I will take a look.

**Comment by megonemad1** [comment]
Gunna bump this, but with a focus on real life examples, I see the usecase and how it all should work and the examples are good but slightly basic. If there is a list of other projects that use it or some more detailed examples to show how best to make tests for games that aren't naughts and crosses. I think that would really help Invision how test driven godot projects should look.
Thanks

**Reply by MikeSchulze** [reply]
I had no time to add advanced examples.
Please check the current tutorials https://mikeschulze.github.io/gdUnit4/tutorials/tdd/#test-driven-development-tdd-with-gdunit4
